pkg/auth/handler/oauth/metadata.go

```diff
@@ -0,0 +1,46 @@
+package oauth
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+
+	pkg "github.com/skygeario/skygear-server/pkg/auth"
+)
+
+func AttachMetadataHandler(
+	router *mux.Router,
+	authDependency pkg.DependencyMap,
+) {
+	handler := pkg.MakeHandler(authDependency, newMetadataHandler)
+	router.NewRoute().
+		Path("/.well-known/openid-configuration").
+		Handler(handler).
+		Methods("GET")
+	router.NewRoute().
+		Path("/.well-known/oauth-authorization-server").
+		Handler(handler).
+		Methods("GET")
+}
+
+type oauthMetadataProvider interface {
+	PopulateMetadata(meta map[string]interface{})
+}
+
+type MetadataHandler struct {
+	metaProviders []oauthMetadataProvider
+}
+
+func (h *MetadataHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	meta := map[string]interface{}{}
+	for _, provider := range h.metaProviders {
+		provider.PopulateMetadata(meta)
+	}
+
+	encoder := json.NewEncoder(rw)
+	err := encoder.Encode(meta)
```

Comment:
It's better to include content-length.

Reply:
I just tried and the Content-Length is also set automagically here...